docs/dev/table/streaming/match_recognize.md

```diff
@@ -0,0 +1,843 @@
+last one started. This would produce an infinite loop and, thus, is prohibited.
+
+One has to keep in mind that in case of the `SKIP TO FIRST/LAST variable`strategy it might be possible that there are no rows mapped to that
+variable (e.g. for pattern `A*`). In such cases, a runtime exception will be thrown as the standard requires a valid row to continue the
+matching.
+
+
+### Controlling Memory Consumption
+
+Memory consumption is an important aspect when writing `MATCH_RECOGNIZE` queries as the space of potential matches is built in a breadth-first like manner.
+Having that in mind, one must make sure that the pattern can finish. Preferably with a reasonable number of rows mapped to the match as they have to fit into memory.
+
+For example, the pattern must not have a quantifier without an upper limit that accepts every single row. Such a pattern could look like this:
+
+{% highlight sql %}
+PATTERN (A B+ C)
+DEFINE
+  A as A.price > 10,
+  C as C.price > 20
+{% endhighlight %}
+
+The query will map every incoming row to the `B` variable and thus will never finish. This query could be fixed, e.g., by negating the condition for `C`:
+
+{% highlight sql %}
+PATTERN (A B+ C)
+DEFINE
+  A as A.price > 10,
+  B as B.price <= 20,
+  C as C.price > 20
+{% endhighlight %}
+
+Or by using the [reluctant quantifier](#greedy--reluctant-quantifiers):
+
+{% highlight sql %}
+PATTERN (A B+? C)
+DEFINE
+  A as A.price > 10,
+  C as C.price > 20
+{% endhighlight %}
+
+<span class="label label-danger">Attention</span> Please note that the `MATCH_RECOGNIZE` clause does not use a configured [state retention time](query_configuration.html#idle-state-retention-time). As of now, there is also no possibility to define a time restriction on the pattern to finish because there is no such possibility in the SQL standard. The community is in the process of designing a proper syntax for that
+feature right now.
+
+Known Limitations
+-----------------
+
+`MATCH_RECOGNIZE` clause is still an ongoing effort and therefore some SQL standard features are not supported yet.
+
+The list of such features includes:
+* Pattern expressions:
+  * Pattern groups - this means that e.g. quantifiers can not be applied to a subsequence of the pattern. Thus, `(A (B C)+)` is not a valid pattern.
+  * Alterations - patterns like `PATTERN((A B | C D) E)`, which means that either a subsequence `A B` or `C D` has to be found before looking for the `E` row.
+  * `PERMUTE` operator - which is equivalent to all permutations of variables that it was applied to e.g. `PATTERN (PERMUTE (A, B, C))` = `PATTERN (A B C | A C B | B A C | B C A | C A B | C B A)`.
+  * Anchors - `^, $`, which denote beginning/end of a partition, those do not make sense in the streaming context and will not be supported.
+  * Exclusion - `PATTERN ({- A -} B)` meaning that `A` will be looked for but will not participate in the output. This works only for the `ALL ROWS PER MATCH` mode.
```

Comment:
Are you saying that exclusion only makes sense in the ALL ROWS PER MATCH mode, and Flink doesn't support exclusion because Flink doesn't support ALL ROWS PER MATCH -- or that Flink currently only supports exclusion in ALL ROWS PER MATCH mode?

Reply:
I see this becomes clearer about two lines later ...